### PR TITLE
Nfdiv 1352 service application tab formatting

### PR DESCRIPTION
**Pull request by mark-butler-solirius**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1352

### Change description ###
Change the archival of the ServiceApplication to create a new Object (ServiceApplicationOutcome). This new object will be displayed for the "Previous Service Applications" and will support the new DisplayOrder field so that the NFD formatting resembles the Divorce1 formatting 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    private YesOrNo successfulServedByBailiff;
+
+    @CCD(
+        label = "Reason for failure to serve",
+        typeOverride = TextArea,
+        displayOrder = "15"
+    )
+    private String reasonFailureToServeByBailiff;
+
+    public String getServiceApplicationOutcomeLabel() {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java:131]: does this need an @JsonIgnore or should it be included?

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class AlternativeServiceOutcome {
+
+    @CCD(
+        label = "Application date",
+        displayOrder = "1"
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java:26]: should these be integers?

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java [hunk 3]
```diff
@@ -364,4 +366,24 @@ public void archiveAlternativeServiceApplicationOnCompletion() {
         }
     }
 
+    @SuppressWarnings("PMD")
+    public AlternativeServiceOutcome buildAlternativeServiceOutcome(AlternativeService alternativeService) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java:370]: it looks to me like this method should be on AlternativeService something like alternativeService.getOutcome()

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java (hunk 2 above)
Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java:26]: Yeah. I initially created the displayOrder attribute as an Integer in the CCD Interface class, but I couldn't initially work out how to get the default value in the Interface to be null. I'm happy to go back and change it, or come up with another solution if we cannot use null defaults.

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java (hunk 3 above)
Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseData.java:370]: Yep - will move it

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java (hunk 2 above)
Comment (linusnorton) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java:26]: assuming it defaults to 0 we could treat that is null

Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java:26]: Ok - will implement this solution

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeService.java [hunk 4]
```diff
@@ -108,4 +109,25 @@
     @CCD(access = {CaseworkerAccessBetaOnlyAccess.class})
     private Bailiff bailiff = new Bailiff();
 
+    @SuppressWarnings("PMD")
+    @JsonIgnore
+    public AlternativeServiceOutcome getOutcome() {
+        return AlternativeServiceOutcome.builder()
+            .alternativeServiceType(this.getAlternativeServiceType())
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeService.java:116]: is there any need to use getters here? maybe just use the local property

Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeService.java:116]: Yes, I can change to the local properties (didn't think about it after I moved this method from the CaseData class)

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java (hunk 1 above)
Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/AlternativeServiceOutcome.java:131]: Yes - this should be included so the Label gets added to the AlternativeServiceOutcome list object

**Review by linusnorton — APPROVED** [review]

**Pull request closed without merging**